backend/routes/user.py

```diff
@@ -46,4 +51,35 @@ async def create_account(user: CreateUserSchema= Body(...)):
 
     return {'status_code':200, 'message': 'User saved successfully'}
 
+# Use OAuth2PasswordBearer for authorization
+oauth2_scheme = OAuth2PasswordBearer(tokenUrl="/api/v1/verify_otp")
+
+@router.post("/api/v1/verify_otp")
+async def verify_otp(vphone_number: str, votp: int, is_creation: bool):
+
+    # Compare the provided OTP with the stored OTP
+    if database.otp_mapping.find_one({'phone_number': vphone_number, 'otp': votp}):
+        # check expiration of entered OTP
+        curr_time = datetime.datetime.now()
+        otp_gen_time = database.otp_mapping.find_one({'phone_number': vphone_number, 'otp': votp},
+                                                     {'timestamp': 1, '_id': 0})
+        valid_time = otp_gen_time['timestamp'] + datetime.timedelta(seconds=60)
+        if curr_time > valid_time:
+            return ("OTP expired")
```

Comment:
Instead of returning this tuple, can return something similar to VerifyOTPResponse, example ErrorResponse.